homeassistant/components/derivative/sensor.py

```diff
@@ -334,19 +339,27 @@ def on_state_changed(event: Event[EventStateChangedData]) -> None:
             schedule_max_sub_interval_exceeded(new_state)
             old_state = event.data["old_state"]
             if old_state is not None:
-                calc_derivative(new_state, old_state.state, old_state.last_reported)
+                calc_derivative(
+                    new_state,
+                    old_state.state,
+                    new_state.last_updated,
+                    old_state.last_reported,
```

Comment:
The old state will not be changed, which means it's safe to use the `last_reported` timestamp